config/config.go

```diff
@@ -100,17 +101,88 @@ type Config struct {
 	StmtSummary         StmtSummary `toml:"stmt-summary" json:"stmt-summary"`
 }
 
+// nullableBool defaults unset bool options to unset instead of false, which enables us to know if the user has set 2
+// conflict options at the same time.
+type nullableBool struct {
+	Valid bool `toml:"valid" json:"valid"`
+	Value bool `toml:"value" json:"value"`
```

Comment:
> Since we implement `UnmarshalText` and `UnmarshalJSON`, we do not need the tags for toml and json here.

Nope, tags are for `marshal`, to follow the previous practice, we want the keys to be lowercase in marshaled data, so I put the tags here.

Reply:
In which case we need marshal `nullableBool`?

Reply:
No, you need, in `http_request.go`, accessing `/settings` will return a JSON encoded configuration file, in this case marshal to JSON is required, although not neccessary for marshal TOML.